app/models/manageiq/providers/vmware/cloud_manager/vm/operations.rb

```diff
@@ -1,3 +1,20 @@
 module ManageIQ::Providers::Vmware::CloudManager::Vm::Operations
+  extend ActiveSupport::Concern
+
   include_concern 'Power'
+
+  included do
+    supports :terminate do
+      unsupported_reason_add(:terminate, "The VM is powered on") unless current_state == "off"
+    end
+  end
+
+  def raw_destroy
+    raise "VM has no #{ui_lookup(:table => "ext_management_systems")}, unable to destroy VM" unless ext_management_system
+    ext_management_system.with_provider_connection do |service|
+      response = service.delete_vapp(ems_ref)
```

Comment:
Instead of checking for exceptions we are simply passing to the underlying provider connection. Looking at OpenStack and Amazon, this is exactly how these two are handled. In case an exception occurs it will be propagated.

Reply:
:+1: perfect